src/basic/fd-util.c

```diff
@@ -427,7 +427,7 @@ int move_fd(int from, int to, int cloexec) {
 
 int acquire_data_fd(const void *data, size_t size, unsigned flags) {
 
-        char procfs_path[strlen("/proc/self/fd/") + DECIMAL_STR_MAX(int)];
+        char procfs_path[STRLEN("/proc/self/fd/") + DECIMAL_STR_MAX(int)];
```

Comment:
Should this have +1 space for '\0'? Somewhere the space is added, but somewhere not... I have not checked the following logic completely, but I am very confused now...

Reply:
DECIMAL_STR_MAX already contains that. Which is confusing. And we should clean that up: make DECIMAL_STR_MAX() return what#s really needed, and then add the +1 in the callers systematically. But for this PR that's really out of scope

Reply:
(cleaning that up is nasty btw, since part of the uses of DECIMAL_STR_MAX() currently are written in knowledge of the implicit +1 it does, and other parts are not. Hence correcting its definition means looking at every single use and adjusting the +1 there or possibly not...

Reply:
Thank you for the explanation. So, now at least these without +1 does not cause segfault. I am very happy to hear that :-)